src/main/java/dev/hephaestus/atmosfera/client/sound/modifiers/implementations/PercentBiomeModifier.java

```diff
@@ -4,30 +4,34 @@
 import com.google.common.collect.ImmutableList;
 import com.google.gson.JsonArray;
 import com.google.gson.JsonObject;
+import com.mojang.datafixers.util.Pair;
 import dev.hephaestus.atmosfera.client.sound.modifiers.AtmosphericSoundModifier;
 import dev.hephaestus.atmosfera.world.context.EnvironmentContext;
-import net.fabricmc.fabric.api.tag.TagFactory;
-import net.minecraft.tag.Tag;
+import net.minecraft.tag.TagKey;
 import net.minecraft.util.Identifier;
 import net.minecraft.util.JsonHelper;
 import net.minecraft.util.registry.Registry;
+import net.minecraft.util.registry.RegistryEntry;
 import net.minecraft.world.World;
 import net.minecraft.world.biome.Biome;
 
-public record PercentBiomeModifier(float min, float max, ImmutableCollection<Biome> biomes, ImmutableCollection<Tag.Identified<Biome>> biomeTags, ImmutableCollection<Biome.Category> biomeCategories) implements AtmosphericSoundModifier {
-    public PercentBiomeModifier(float min, float max, ImmutableCollection<Biome> biomes, ImmutableCollection<Tag.Identified<Biome>> biomeTags, ImmutableCollection<Biome.Category> biomeCategories) {
-        ImmutableCollection.Builder<Biome> biomesBuilder = ImmutableList.builder();
+public record PercentBiomeModifier(float min, float max, ImmutableCollection<Pair<Biome, RegistryEntry<Biome>>> biomes, ImmutableCollection<TagKey<Biome>> biomeTags, ImmutableCollection<Biome.Category> biomeCategories) implements AtmosphericSoundModifier {
+    public PercentBiomeModifier(float min, float max, ImmutableCollection<Pair<Biome, RegistryEntry<Biome>>> biomes, ImmutableCollection<TagKey<Biome>> biomeTags, ImmutableCollection<Biome.Category> biomeCategories) {
+        ImmutableCollection.Builder<Pair<Biome, RegistryEntry<Biome>>> biomesBuilder = ImmutableList.builder();
 
-        // Remove blocks that are already present in tags so that they aren't counted twice
+        // Remove biomes that are already present in tags so that they aren't counted twice
         biomes:
-        for (Biome biome : biomes) {
-            for (Tag<Biome> tag : biomeTags) {
-                if (tag.contains(biome) || biomeCategories.contains(biome.getCategory())) {
+        for (Pair<Biome, RegistryEntry<Biome>> biomePair : biomes) {
+            Biome biome = biomePair.getFirst();
+            RegistryEntry<Biome> biomeEntry = biomePair.getSecond();
+
+            for (TagKey<Biome> tag : biomeTags) {
+                if (biomeEntry.isIn(tag) || biomeCategories.contains(Biome.getCategory(biomeEntry))) {
```

Comment:
In contrast to `Block`/`BlockState`, `Biome` has neither `isIn()` nor `getRegistryEntry()`, hence this was rewritten to accept a biome together with its `RegistryEntry`, see `create()` below.